### ISSUE TITLE
manual_abs_unsigned

### ISSUE THREAD

**Issue by owenthewizard** [rust-lang/rust-clippy#13786]
### What it does

Replace `-{signed} as {unsigned}` with `{signed}.abs_unsigned()`

### Advantage

Clear and concise meaning & order of operations.
No panics or overflow (potentially).

### Example

```rust
let n: isize = -42;
-n as usize
```

Could be written as:

```rust
let n: isize = -42;
n.abs_unsigned()
```

**Comment by samueltardieu** [comment]
The example works only because `n` is negative. Should `n` be positive, the result would not be the same at all.

If you are thinking about `n.abs() as usize`, this is already handled by the `CAST_ABS_TO_UNSIGNED` lint.

**Comment by owenthewizard** [comment]
> The example works only because `n` is negative. Should `n` be positive, the result would not be the same at all.
> 
> If you are thinking about `n.abs() as usize`, this is already handled by the `CAST_ABS_TO_UNSIGNED` lint.

Yeah, I accidentally omitted it from my example but in my code I had this:
```rust
let n: isize = -42;
if n.is_negative() {
    -n as usize
}
```
